### DISCUSSION TITLE
Pie Charts - Ordering Option

### DISCUSSION THREAD

**Discussion by katiebuilds** [mermaid-js/mermaid#5203] [Ideas]
TL;DR: would it be possible to have a configuration option as to whether you want the pie chart to be ordered in descending order based on the values, or ordered the way you typed it in, as it used to be?

I use Mermaid in my budget page in Notion, because they don't support graphs/charts, and I don't want to enter this kind of data somewhere else and embed it.

I like to see pie charts of how I'm spending my money as far as Needs, Wants, and Savings.  I have charts for the budgeted amount, year-to-date totals, and each month.

It used to be that the pie chart was in the order you typed it in, which worked great for me.
* Here was an issue requesting this from 2022: https://github.com/mermaid-js/mermaid/issues/3593
* And the fix: https://github.com/mermaid-js/mermaid/pull/3609

But it seems like a fairly recent bugfix set the order to sort by values:
* Issue: https://github.com/mermaid-js/mermaid/issues/4772
* Fix: https://github.com/mermaid-js/mermaid/pull/4774

So now, it makes it really hard for me to compare my budget to my actual.  Savings is the largest percentage in the budget, but often, Wants is the biggest category for what I actually spend.  😊  The different order makes the colors no longer correspond:
![image](https://github.com/mermaid-js/mermaid/assets/10214426/9681735d-37f7-4163-8d88-8e753b604d06)

![image](https://github.com/mermaid-js/mermaid/assets/10214426/f3ec72ac-d8a6-456b-b4f7-17e23ee3e8b8)

The [documentation](https://mermaid.js.org/syntax/pie.html) still says it's ordered by label:
<img width="767" alt="image" src="https://github.com/mermaid-js/mermaid/assets/10214426/6c126966-00b1-4856-b429-eb5e2c41b507">

even though this is no longer the case:
<img width="784" alt="image" src="https://github.com/mermaid-js/mermaid/assets/10214426/fbb01a95-3b18-4aa8-8b99-ad99dffa5cef">

I personally would be happy with things back the way they were, where the order you typed it in was the order of the chart.  But perhaps other people prefer it ordered in descending order automatically, so maybe it would be nice to have a configuration option where you could specify descending or the order you typed?

**Comment by Octobug** [comment]
I also need this. When there is an "other" item, I would like to have it as the last one no matter how big its value is.

**Comment by BPowell76** [comment]
The documentation mentions a `dataSet` parameter, but I cannot figure out how to use it. I do wish I could order the graph. It makes setting custom colors kind of pointless if the area being filled isn't the same category.
